src/web/mjs/connectors/MangaFox.mjs

```diff
@@ -94,58 +97,38 @@ export default class MangaFox extends Connector {
     }
 
     async _getPages(chapter) {
-        return this._getPagesDesktop(chapter);
-        //return this._getPageListMobile(chapter);
-    }
-
-    async _getPagesDesktop(chapter) {
-        let uri = new URL(chapter.id, this.url);
+        const uri = new URL(chapter.id, this.url);
         let request = new Request(uri, this.requestOptions);
+        const pages = await Engine.Request.fetchUI(request, this.script);
 
-        let pages = await Engine.Request.fetchUI(request, this.script);
-        //Get Meta informations from last Image
-        let lastImage = await new Promise(resolve => {
-            const img = new Image();
-            img.onload = () => resolve(img);
-            img.onerror = () => resolve(null);
-            img.src = pages.slice(-1);
-        });
-        //Check if the last image seems to be the usual ad
-        if(lastImage && lastImage.naturalHeight === 563 && lastImage.naturalWidth === 1000) {
-            pages.pop();
+        try {
+        //fetch last image data
+            request = new Request(pages.slice('-1'));
+            request.headers.set('x-referer', uri);
+            const response = await fetch(request);
```

Comment:
Similar code as in `this. _handleConnectorURI`, suggestion to use the existing method:
```javascript
const pageList = pages.map(link => this.createConnectorURI({
    url: link,
    referer: this.url
}));

const lastImage = await this. _handleConnectorURI(pageList.slice(-1));
// perform advertisement check and may remove last image: pageList.pop()

return pageList;
```